libnetwork/netavark/config.go

```diff
@@ -125,6 +125,17 @@ func (n *netavarkNetwork) networkCreate(newNetwork *types.Network, defaultNet bo
 		if err != nil {
 			return nil, err
 		}
+	case types.WireGuardNetworkDriver:
+		// The network is purely created so that containers can join it
+		// with their actual interface configuration
+		// To disable any Gateway generation or DNS support the network
+		// is set to internal and DNS forcefully disabled
+		err = createMockWireGuardNetwork(newNetwork)
+		newNetwork.DNSEnabled = false
+		newNetwork.Internal = true
```

Comment:
This is a *hack*. These fields are exposed in podman network inspect and IMO make no sense to mark it as internal?

AFAIK the call below should not matter since no subnets are used, therefore the ValidateSubnets call does nothing.
Instead you should check here that subnets and gateways are not set for a wireguard network.

Reply:
Removed this and added check for subnets